### PR TITLE
ml: refactored tests

### PR DESCRIPTION
**Pull request by mshabunin**
### This pullrequest changes

* reorganized test files
* use parametrized tests where appropriate
* use stable theRNG in most tests
* use modern style with EXPECT_/ASSERT_ checks

Notes:
* `KNearest/KDTree` is broken - does not initialize output (this problem was previously shadowed by preceding test code).
* `opencv_extra/testdata/ml/{a,sl}validataion.xml` can be removed (I'll clean them later)

### CODE REVIEW

File: modules/ml/test/test_bayes.cpp [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+TEST(ML_NBAYES, regression_5911)
+{
+    int N=12;
+    Ptr<ml::NormalBayesClassifier> nb = cv::ml::NormalBayesClassifier::create();
+
+    // data:
+    Mat_<float> X(N,4);
+    X << 1,2,3,4,  1,2,3,4,   1,2,3,4,    1,2,3,4,
+         5,5,5,5,  5,5,5,5,   5,5,5,5,    5,5,5,5,
+         4,3,2,1,  4,3,2,1,   4,3,2,1,    4,3,2,1;
```
Comment (alalek) [inline modules/ml/test/test_bayes.cpp]: Lets avoid bloating of binary code and slowdown compiler to generate that.

```
    float X_data[] = {
        1,2,3,4,  1,2,3,4,   1,2,3,4,    1,2,3,4,
        5,5,5,5,  5,5,5,5,   5,5,5,5,    5,5,5,5,
        4,3,2,1,  4,3,2,1,   4,3,2,1,    4,3,2,1
    };
    Mat_<float> X(N,4, X_data);
```

(58Kb -> 50Kb for .o file: 1.7Kb of filling code + exception tables and other code helpers)

File: modules/ml/test/test_ann.cpp [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+#include "test_precomp.hpp"
+
+// #define GENERATE_TESTDATA
+
+namespace opencv_test { namespace {
+
+struct Activation
+{
+    int id;
+    string name;
```
Comment (alalek) [inline modules/ml/test/test_ann.cpp]: `const char*`

File: modules/ml/test/test_save_load.cpp [hunk 3]
```diff
@@ -271,33 +104,32 @@ TEST(ML_SVMSGD, legacy_load) { CV_LegacyTest test(CV_SVMSGD, "_waveform.xml"); t
...
-
-    double eps = 1e-4;
-    EXPECT_LE(cvtest::norm(r1, r2, NORM_INF), eps);
-    EXPECT_LE(cvtest::norm(r1, r3, NORM_INF), eps);
-
-    remove(tname.c_str());
-}
+// TEST(DISABLED_ML_SVM, linear_save_load)
+// {
+//     Ptr<cv::ml::SVM> svm1, svm2, svm3;
```
[the remark this answers is not in the document]
Comment (mshabunin) [reply modules/ml/test/test_save_load.cpp]: This test uses XML files which have never been committed anywhere. I doubt we will be able to run it.

Removed.

File: modules/ml/test/test_kmeans.cpp [hunk 4]
```diff
@@ -0,0 +1,53 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "test_precomp.hpp"
+
+namespace opencv_test { namespace {
+
+TEST(ML_KMeans, accuracy)
```
[the remark this answers is not in the document]
Comment (mshabunin) [reply modules/ml/test/test_kmeans.cpp]: This test uses test data generator from _ml_ module. We can move this generator to _ts_  and the test to _core_ module. But I think it is not in the scope of this PR.

**Pull request merged**